Changed register route and waiver routing

ryanchansf opened the pull request:
Changed register route to check for duplicate volunteers and new volunteers. Changed routing to match whether the returned user signedWaiver or not. Made sure no duplicate volunteers were added to specific Event (note: didn't throw any error on frontend but may have to be added later to avoid confusion for the user)

ryanchansf committed:
Changed register route and waiver routing

Changed register route to check for duplicate volunteers and new volunteers. Changed routing to match whether the returned user signedWaiver or not. Made sure no duplicate volunteers were added to specific Event (note: didn't throw any error on frontend but may have to be added later to avoid confusion for the user)

vwinstea commented on frontend/src/components/HomePage/home.jsx:
```diff
@@ -233,7 +229,14 @@ export default function Home({ selectedEvent, setEvent }) {
           </Route>
           <Route path="/waiver">
-            <WaiverPage user={user} />
+            {user.signedWaiver ? (
+              <RegistrationComplete
+                selectedEvent={selectedEvent}
+                handleModalClose={handleModalClose}
+              />
+            ) : (
+              <WaiverPage user={user} />
+            )}
```
Could you change where the link on the SignUp page goes instead of changing this route dynamically? Typically it's not the best practice to have the route change dynamically like this.

vwinstea commented on backend/routes/volunteer.js:
```diff
@@ -65,21 +66,42 @@ router.post(
+        });
+      } else {
+        res.json(volunteer);
+        // eslint-disable-next-line no-throw-literal
+        // throw {
+        //   message: "Volunteer array already contains user for this event",
+        // };
```
Right now the backend crashes when a user signs up for an event they're already signed up for. Could you handle this edge case?

vwinstea commented on frontend/src/components/UserSignUp/SignUpForm.jsx:
```diff
@@ -223,11 +223,15 @@ function SignUpForm({ selectedEvent, handleModalClose, isAdult }) {
-    reset();
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
```
Could you delete this console.log?

vwinstea commented on frontend/src/components/WaiverComponent/WaiverPage.jsx:
```diff
@@ -107,7 +107,8 @@ const WaiverExplanation = styled.p`
 export default function WaiverPage({ user }) {
   const signWaiver = () => {
-    fetch(`/volunteer/${user.id}/signWaiver`, {
+    console.log(user);
```
Could you delete this console.log?

vwinstea committed:
Merge branch 'main' of https://github.com/hack4impact-calpoly/city-farm-slo into ryan-chan-repeat-user-routing

vwinstea reviewed (changes requested):
Great work! There's a couple small changes I think we should make before merging this in, but overall its looking great! Thanks for doing this, it was not an easy task.

ryanchansf committed:
Updated routing

Modified routing, got rid of backend crash for repeat user for event, got rid of unnecessary console.logs

ryanchansf committed:
Merge branch 'ryan-chan-repeat-user-routing' of https://github.com/hack4impact-calpoly/city-farm-slo into ryan-chan-repeat-user-routing

vwinstea reviewed (commented):
Looks good!
